extension/parse-word-dict-entry.ts

```diff
@@ -0,0 +1,21 @@
+const WORD_DICT_ENTRY_REGEX = /^(.+?)\s+(?:\[(.*?)\])?\s*\/(.+)\//;
+
+type WordDictEntry = {
+  word: string;
+  pronunciation: string;
+  definitions: string[];
+};
+
+export const parseWordDictEntry = (line: string): WordDictEntry | null => {
+  const [, word, pronunciation, definitions] =
+    line.match(WORD_DICT_ENTRY_REGEX) || [];
+  if (!word || !definitions) {
+    console.log('Improperly formatted word dict line', line);
```

Comment:
The downside of having a console.log here is it shows up in the test output. I'm wondering if I should silence it somehow for tests.

Reply:
Is there a problem for it to show up in the test output? In fact, adding a
new test for the log would make sense to me since its an external side
effect

I assume it only shows up only in the tests which are triggering the
malformed branch?

Reply:
I guess it just looks kind of weird when I see that the tests all pass but the logs say "Improperly formatted word dict line". It makes me feel like something is wrong and the tests aren't catching it even though it's logging as expected. It seems like it clutters the logs a little bit, but it's not a big deal.

Yeah, it only shows up in tests which are triggering the malformed branch.

Reply:
I added a test which checks that the console.log function is called for malformed dict lines.